trunk/modules/python/dataset/ingest/labeled.py

```diff
@@ -101,6 +105,78 @@ def readImagery() :
     # join the threads and complete
     workQueueData.join()
 
+
+def hdf5get(inputfile, key, metadata=False):
+    ''' Get a record from an hdf5 file '''
+    with h5py.File(inputfile, 'r') as f:
+        value = f[key]
+        if metadata:
+            rval = value[:], dict(value.attrs)
+        else:
+            rval = value[:]
+        return rval
+
+
+class FileData:
+    """Filesystem-backed file data provider
+
+    :param rootpath: path to directory of files 
+    """
+    def __init__(self, rootpath):
+        self.rootpath = rootpath
+
+    def walk(self):
+        return os.walk(self.rootpath)
+
+    def getGetSize(self):
+        return lambda f: os.path.getsize(f)
+
+    def getImageShape(self, filename, log=None):
+        from dataset.reader import getImageDims
+        return list(getImageDims(filename, log))
+
+    def readImage(self, filename, log=None):
+        from dataset.reader import readImage
+        return readImage(filename, log)
+
+
+class Hdf5FileData(FileData):
+    """Hdf5-backed file data provider
+
+    :param rootpath: path to hdf5 file
+    """
+    def __init__(self, rootpath):
+        self.rootpath = rootpath
+
+    def walk(self):
+        from dataset.hdf5walk import hdf5walk
+        return hdf5walk(self.rootpath)
+
+    def getGetSize(self):
+        from functools import partial
+        def h5filesize(inputfile, key):
+            return hdf5get(inputfile, key).nbytes
+        return partial(h5filesize, self.rootpath)
+
+    def getImageShape(self, filename, log=None):

```

Comment:
This would result in a recursive loop, since getImageDims calls readImage; But we need to call readImage with a tuple in the current setup so that it knows how to interpret the np array returned from the hdf5 (may be OBE after we do the open* refactoring)